proc-macros/tests/ui/correct/basic.rs

```diff
@@ -24,6 +24,13 @@ pub trait Rpc {
 	#[method(name = "array_params")]
 	async fn array_params(&self, items: Vec<u64>) -> RpcResult<u64>;
 
+	#[method(name = "rename_params")]
+	async fn rename_params(
+		&self,
+		#[argument(rename = "type")] r#type: u16,
+		#[argument(rename = "halfType")] half_type: bool,
```

Comment:
I don't recall here either; would `halfType` not already be accepted or do we not to any camel cas transform?

I'd feel more confident that the test was working if the rename was to something more different than the original :)

Reply:
You are right for servers generated by jsonrpsee, they accept mutliple cases ([see here](https://github.com/paritytech/jsonrpsee/blob/master/proc-macros/src/render_server.rs#L390)).

When talking to other JSON-RPC servers this might not be the case, hence my PR.